### ISSUE TITLE
Received status code 403 from server

### ISSUE THREAD

**Issue by Vishal949** [mapbox/mapbox-navigation-android#4372]
**Android API:** Min 21  to  Max 29
**Mapbox Navigation SDK version:** 0.4.0

### Steps to trigger behavior
Could not GET 'https://api.mapbox.com/downloads/v2/releases/maven/com/mapbox/gradle/plugins/sdk-registry/0.4.0/sdk-registry-0.4.0.jar'. Received status code 403 from server

Setup all specification regaring included in document also i am facing this iisue in may project,Please give solution for set secrate key in MAPBOX_DOWNLOADS_TOKEN  and SDK_REGISTRY_TOKEN
### Expected behavior
allprojects {
    repositories {
        google()
        mavenCentral()
        jcenter()
        maven { url "https://jitpack.io" }
        maven { url 'https://plugins.gradle.org/m2' }
        maven {
            url 'https://api.mapbox.com/downloads/v2/releases/maven'
            authentication {
                basic(BasicAuthentication)
            }
            credentials {
                username = "mapbox"
                password = project.hasProperty('SDK_REGISTRY_TOKEN') ? project.property('SDK_REGISTRY_TOKEN') : System.getenv('SDK_REGISTRY_TOKEN')
            }
        }
        maven { url 'https://mapbox.bintray.com/mapbox' }
    }
}

### Actual behavior
Download Sdk build may project....

**Comment by LukasPaczos** [comment]
Hey @Vishal949, in order to build the project you need to set up credentials the same way as described in https://docs.mapbox.com/android/beta/navigation/guides/install/ where the `password` field is your token.

**Comment by jeankruger** [comment]
Hello @LukasPaczos I have the same issue. With curl I've this error message:
`{"message":"This is a prerelease API. Please contact support at help@mapbox.com to request access."}`

**Comment by LukasPaczos** [comment]
@jeankruger which version of the Navigation SDK are you trying to access?

**Comment by jeankruger** [comment]
I've tested from the master branch and from branch based on the tag `1.6.1`

**Comment by sandyscoffable** [comment]
Hit this today. It's annoying as you can't run the example demo app.

**Comment by jeankruger** [comment]
Yes, and we can't make improvements / fix issues / propose a PR neither

**Comment by LukasPaczos** [comment]
I'm able to reproduce the issue, we are tracking down why our Maven repository doesn't want to serve that artifact.

**Comment by jkpatel1463** [comment]
I have purchased support plan to get answer from support team for this issues. I created support ticet May 10, 2021, Still i am waiting for them to answer my support ticket. Support ticket  #105868

**Comment by LukasPaczos** [comment]
Thanks for your patience @jkpatel1463, we are looking into this issue.

**Comment by jkpatel1463** [comment]
Is there any updates on it? We stuck on this point and we are not compiling our application even. We are not able to deliver updates of the app even for this issues. Its more critical issues for us.

**Comment by LukasPaczos** [comment]
No updates just yet, but this shouldn't impact consuming regular Nav SDK artifacts, only building this repo from source.

@jkpatel1463 are you not able to build your own app with instructions from https://docs.mapbox.com/android/navigation/guides/install/? Could you share your gradle setup?

**Comment by jkpatel1463** [comment]
Hello @LukasPaczos 

Appreciated and thanks for the update!! We need some customization in the design of the mapbox, So, We use source of the navigation repo & integrate it with our app.

**Comment by LukasPaczos** [comment]
Could you share what is the customization that you're looking for, or cut a separate issue? We'd love to consider that as a future feature so that you can take advantage of the regular artifacts instead.

**Comment by jkpatel1463** [comment]
Hello @LukasPaczos 

Here are the list of customization we did in the navigatrion SDK:
1. Custom Pin set for destination
2. Change icon of the current location (Triangle with Car)
3. Updated bottom view of the screen with ride information
4. Updated top bar view

Here i have attached a link with all customization we did:

https://www.dropbox.com/s/6a59mr4c38l46vx/Mapbox-customization.png?dl=0

Thank you

**Comment by LukasPaczos** [comment]
Thanks for the context @jkpatel1463!
cc @AhmerKhan1 @jyrigo @abhishek1508 @d-prukop 

Going back to the original issue, this is something we're still looking into.

Meanwhile, for anyone blocked by this, you can work around the issue by removing [this plugin](https://github.com/mapbox/mapbox-navigation-android/blob/bf9f271f309c4d704d21dce0ff4389486170b4f7/build.gradle#L27) and files that depend on it. You should be able to successfully build the project without them. An example diff:
```diff
diff --git a/build.gradle b/build.gradle
index 0e49e3f50..f4eae07d4 100644
--- a/build.gradle
+++ b/build.gradle
@@ -24,7 +24,6 @@ buildscript {
     classpath pluginDependencies.googleServices
     classpath pluginDependencies.license
     classpath pluginDependencies.mapboxSdkVersions
-    classpath pluginDependencies.mapboxSdkRegistry
     classpath pluginDependencies.mapboxAccessToken
     classpath pluginDependencies.mapboxNativeDownload
     classpath pluginDependencies.dokka
diff --git a/gradle/publish.gradle b/gradle/publish.gradle
index a43f218c2..b1e0e6168 100644
--- a/gradle/publish.gradle
+++ b/gradle/publish.gradle
@@ -1,5 +1,5 @@
 apply plugin: 'maven-publish'
-apply plugin: 'com.mapbox.sdkRegistry'
+//apply plugin: 'com.mapbox.sdkRegistry'
 apply from: file("${rootDir}/gradle/artifact-settings.gradle")
 apply from: "${rootDir}/gradle/kdoc-settings.gradle"
 
@@ -67,7 +67,7 @@ sdkNameMap["libnavui-resources"] = "mobile-navigation-ui-resources"
 sdkNameMap["libnavui-voice"] = "mobile-navigation-ui-voice"
 sdkNameMap["libnavui-speedlimit"] = "mobile-navigation-ui-speedlimit"
 
-registry {
+/*registry {
     sdkName = sdkNameMap[project.name]
     production = true
     snapshot = project.ext.versionName.endsWith("-SNAPSHOT")
@@ -75,7 +75,7 @@ registry {
     dryRun = false
     publish = true
     publications = ["release"]
-}
+}*/
 
 task androidSourcesJar(type: Jar) {
     classifier = 'sources'
```

**Comment by jkpatel1463** [comment]
Hello @LukasPaczos 

Okay, Thank you. We will try this one and i will submit result for the same. We may need 24 hours to do this update.

Its curious for me that, Can we remove bottom portion of the mapbox and rest of the data can be retrieved using callback handler of the any mapboxcalss?

If we will have satisfactory anwer that, We can overlay our customize design with mapbox with installation instruction given by you.

**Comment by LukasPaczos** [comment]
> Can we remove bottom portion of the mapbox and rest of the data can be retrieved using callback handler of the any mapboxcalss?

If you're using version 1.x and the `NavigationView`, then no, you can't remove the bottom banner. You'd have to follow https://docs.mapbox.com/android/navigation/guides/choose-an-approach/#collection-of-standalone-components instead. This is the type of customization that we're planning to explore in future iterations though.

**Comment by jkpatel1463** [comment]
> Thanks for the context @jkpatel1463!
> cc @AhmerKhan1 @jyrigo @abhishek1508 @d-prukop
> 
> Going back to the original issue, this is something we're still looking into.
> 
> Meanwhile, for anyone blocked by this, you can work around the issue by removing [this plugin](https://github.com/mapbox/mapbox-navigation-android/blob/bf9f271f309c4d704d21dce0ff4389486170b4f7/build.gradle#L27) and files that depend on it. You should be able to successfully build the project without them. An example diff:
> 
> ```diff
> diff --git a/build.gradle b/build.gradle
> index 0e49e3f50..f4eae07d4 100644
> --- a/build.gradle
> +++ b/build.gradle
> @@ -24,7 +24,6 @@ buildscript {
>      classpath pluginDependencies.googleServices
>      classpath pluginDependencies.license
>      classpath pluginDependencies.mapboxSdkVersions
> -    classpath pluginDependencies.mapboxSdkRegistry
>      classpath pluginDependencies.mapboxAccessToken
>      classpath pluginDependencies.mapboxNativeDownload
>      classpath pluginDependencies.dokka
> diff --git a/gradle/publish.gradle b/gradle/publish.gradle
> index a43f218c2..b1e0e6168 100644
> --- a/gradle/publish.gradle
> +++ b/gradle/publish.gradle
> @@ -1,5 +1,5 @@
>  apply plugin: 'maven-publish'
> -apply plugin: 'com.mapbox.sdkRegistry'
> +//apply plugin: 'com.mapbox.sdkRegistry'
>  apply from: file("${rootDir}/gradle/artifact-settings.gradle")
>  apply from: "${rootDir}/gradle/kdoc-settings.gradle"
>  
> @@ -67,7 +67,7 @@ sdkNameMap["libnavui-resources"] = "mobile-navigation-ui-resources"
>  sdkNameMap["libnavui-voice"] = "mobile-navigation-ui-voice"
>  sdkNameMap["libnavui-speedlimit"] = "mobile-navigation-ui-speedlimit"
>  
> -registry {
> +/*registry {
>      sdkName = sdkNameMap[project.name]
>      production = true
>      snapshot = project.ext.versionName.endsWith("-SNAPSHOT")
> @@ -75,7 +75,7 @@ registry {
>      dryRun = false
>      publish = true
>      publications = ["release"]
> -}
> +}*/
>  
>  task androidSourcesJar(type: Jar) {
>      classifier = 'sources'
> ```

Hello @LukasPaczos 

Thank you very much. Really appreciate your help. We are able to run the Mapbox navigation lib by applying above code. We have to take pull of the mapbox-navigation-git master-branch. but we are able to run the code. 

Thank you

**Comment by LukasPaczos** [comment]
This issue should now be resolved and you should be able to build the project without any modifications to the scripts.

**Comment by jeankruger** [comment]
> This issue should now be resolved and you should be able to build the project without any modifications to the scripts.

Thank you
